### PR TITLE
PAYARA-2584 Don't Let Payara 5 Start if JDK Version is Less than 8u161

### PR DESCRIPTION
**Pull request by MeroRai**

### CODE REVIEW

**Comment by MeroRai** [comment]
Jenkins test please

**Comment by Pandrex247** [comment]
Jenkins test please

**Comment by ggam** [comment]
Could this also apply to JDK 9?

**Comment by smillidge** [comment]
I think the minimum version is 161 not 162

File: nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/MainHelper.java [hunk 1]
```diff
@@ -107,6 +118,16 @@ private static int getMinorJdkVersion() {
         }
     }
 
+    private static int getSecurityJdkVersion() {
+        String jv = System.getProperty("java.version");
```
Comment (arjantijms) [inline nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/MainHelper.java]: Avoid using abbreviations like "jv", just use "javaVersion" or "jdkVersion" etc here.

**Comment by Pandrex247** [comment]
We actually need to udpate the JDK version on Jenkins before this will pass, as it doesn't use the new required version!

File: nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/MainHelper.java (hunk 1 above)
Comment (lprimak) [reply nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/MainHelper.java]: There is also already JDK.java class in nucleus/common to take care of things like this

**Comment by arjantijms** [comment]
Jenkins test

**Comment by smillidge** [comment]
I have a concern with this in that it is actually possible to run on less than JDK 8u161 by switching the Grizzly NPN version.

**Comment by lprimak** [comment]
That's probably true, so maybe we should morph this fix into new version-dependent options in domain templates

**Comment by MeroRai** [comment]
@lprimak Ya, I think it will better to change this fix to new version-dependent options in domain templates

**Pull request closed without merging**